features.md

```diff
@@ -126,3 +126,50 @@ clangd respects your project's `.clang-format` file which controls styling
 options.
 
 (Format-as-you-type is experimental and doesn't work well yet).
+
+
+## Refactoring
+
+### Rename
+
+Rename a symbol under the cursor. All usages of the symbol will be renamed,
+including **declaration**, **definition** and **references**.
+
+![screenshot: rename](screenshots/rename.gif)
+
+Most symbols are renameable, such as classes, variables, functions and methods.
+
+**KNOWN LIMITATIONS**
+- unsupported symbols:
+    namespaces, macros, non-identifier symbols (e.g. overloaded operators),
+    virtual/overriden methods;
+- some references are not renamed (likely need manual modifications):
+  - references inside macro bodies
+  - references in comments
+  - references in generated files
+  - references in platform-specific code
+- templates maybe not perfect (explicit template specializations are missing, [b/280](https://github.com/clangd/clangd/issues/280))
+- no guaranatee on ObjC?
+- limit up to 50 files (to be friendly to editors)
+
+> TIP: the rename experience highly depends on the editor you are using. Some
+> editors, e.g. VSCode (v1.42.0 or later), provide a way to preview the rename
+> changes before applying them; while some just apply the changes directly.
+{:.tip}
+
+#### Within-file rename
+{:.v7}
+
+This is by-default mode, which only allows to rename a local symbol (which is
+only used in current main file).
+
+
+#### Cross-file rename
+{:.v10}
+
+Under an experimental command-line flag `-cross-file-rename`.
+
+The cross-file rename is based on the index, it is fast (no parsing of extra
```

Comment:
It uses the project index(link!) to find references quickly, and so works best when the index is up-to date.

(I don't think "trading off some correctness" or "clangd will try to patch up the results" are things most users need to know, and those who understand this stuff will be able to read between the lines)